src/main/java/revai/StreamingClient.java

```diff
@@ -0,0 +1,205 @@
+   * @param streamContentType
+   * @see StreamContentType
+   */
+  public StreamingClient(String accessToken, StreamContentType streamContentType) {
+    this.accessToken = accessToken;
+    this.streamContentType = streamContentType;
+    this.client = setClient();
+  }
+
+  /**
+   * This method is used for internal testing of the websocket connection against a mock webserver
+   * and should not be used
+   *
+   * @param scheme
+   */
+  public void setScheme(String scheme) {
+    this.scheme = scheme;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  /**
+   * This method is used for internal testing and should not be used
+   *
+   * @param host
+   */
+  public void setHost(String host) {
+    this.host = host;
+  }
+
+  public String getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Setting metadata is optional but helps to identify the stream.
+   *
+   * @param metadata String size restricted to <= 256 characters
+   */
+  public void setMetadata(String metadata) {
+    this.metadata = metadata;
+  }
+
+  public String getCustomVocabularyId() {
+    return customVocabularyId;
+  }
+
+  /**
+   * Setting a custom vocabulary is optional. This method assigns a custom vocabulary to be used
+   * during the stream. Custom vocabularies are submitted prior to usage in the stream and assigned
+   * an Id.
+   *
+   * @param customVocabularyId
+   */
+  public void setCustomVocabularyId(String customVocabularyId) {
+    this.customVocabularyId = customVocabularyId;
+  }
+
+  public Boolean getFilterProfanity() {
+    return filterProfanity;
+  }
+
+  /**
+   * Setting the profanity filter is optional. More info on the expected can be found at
+   * https://www.rev.ai/docs/streaming#section/WebSocket-Endpoint/Filter-Profanity
+   *
+   * @param filterProfanity This is false by default.
+   */
+  public void setFilterProfanity(Boolean filterProfanity) {
+    this.filterProfanity = filterProfanity;
+  }
+
+  /**
+   * Sends an HTTP request and upon authorization is upgraded to a websocket connection. Use
+   * websocket listener to handle websocket events.
+   *
+   * @param revAiWebsocketListener
+   * @see WebSocketListener
+   * @throws URISyntaxException
+   */
+  public void connect(RevAiWebsocketListener revAiWebsocketListener) throws URISyntaxException {
+    Listener listener = new Listener(revAiWebsocketListener);
+    Request request = new Request.Builder().url(buildURL()).build();
+    webSocket = client.newWebSocket(request, listener);
+  }
+
+  /**
+   * Sends data over websocket in the form of a ByteString
+   *
+   * @param byteString
+   */
+  public void sendAudioData(ByteString byteString) {
+    webSocket.send(byteString);
+  }
+
+  /** Will close the websocket connection by informing the server that it's the End of Stream */
+  public void close() {
+    webSocket.send("EOS");
+  }
+
+  private String buildURL() throws URISyntaxException {
+    URIBuilder uriBuilder = new URIBuilder();
```

Comment:
not a huge deal, but it looks like we are taking dependency on org.apache.httpclient library just for this url builder. 
OkHttp already has a url builder: https://square.github.io/okhttp/4.x/okhttp/okhttp3/-http-url/. so i think its better to use that and avoid an extra dependency.

reducing number of dependencies is generally good practice, but httpclient is particularly thorny because often it is included in an appserver like Tomcat itself, and there can be conflict if its also included with an application WAR. i actually ran into this when using your library with Red5 server.